rbac/management/group/definer.py

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Handler for system defined group."""
+import logging
+
+from management.group.model import Group
+from management.policy.model import Policy
+from management.principal.model import Principal
+from management.role.model import Role
+from tenant_schemas.utils import tenant_context
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+def _create_group(tenant, role_list):
+    """Create the group object in the database."""
+    with tenant_context(tenant):
+        name = "Platform group"
+        group_description = "Default group that contains default permissions for new users."
+        policy_description = "Platform default policy that is binded with default group."
+
+        principals = list(Principal.objects.all())
+        group = Group.objects.create(name=name,
+                                     description=group_description,
+                                     system=True)
+        group.principals.add(*principals)
```

Comment:
The concept of system groups is that we don't need to add users to it. Otherwise we have to manage new users, etc. A "system" group should just imply all users have access (existing and all new), so we really need it to behave a bit differently, where all users get this group by default. Essentially `principal.groups` would be all groups for the principal, plus the system group.

Reply:
I see, thanks for clarifying this. Then we should overwrite the principal groups method.